### PR TITLE
auto reporting

### PR DESCRIPTION
**Pull request by wabinyai**
**_WHAT DOES THIS PR DO?_**

- [x] Intending to automate air quality reports by grid
- [x] Intended to have an API that prints data in JSON

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - []
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**
"flask run"
- [x] Analytics: [Link to Analytics README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/analytics#readme)
- [ ] Auth Service: [Link to Auth Service README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/auth-service#readme)
- [ ] Calibrate: [Link to Calibrate README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/calibrate#readme)
- [ ] Data Proxy: [Link to Data Proxy README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/data-proxy#readme)
- [ ] Device Monitoring: [Link to Device Monitoring README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-monitoring#readme)
- [ ] Device Registry: [Link to Device Registry README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-registry#readme)
- [ ] Device Status: [Link to Device Status README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-status#readme)
- [ ] Device Uptime: [Link to Device Uptime README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-uptime#readme)
- [ ] Exceedances: [Link to Exceedances README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/exceedances#readme)
- [ ] Firebase: [Link to Firebase README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/firebase)
- [ ] GP Model: [Link to GP Model README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/gp-model#readme)
- [ ] Incentives: [Link to Incentives README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/incentives#readme)
- [ ] Kafka Connectors: [Link to Kafka Connectors](https://github.com/airqo-platform/AirQo-api/tree/staging/src/kafka-connectors)
- [ ] Locate: [Link to Locate README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/locate#readme)
- [ ] Metadata: [Link to Metadata README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/meta-data#readme)
- [ ] Predict: [Link to Predict README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/predict#readme)
- [ ] Spatial: [Link to Spatial](https://github.com/airqo-platform/AirQo-api/tree/staging/src/spatial)
- [ ] View: [Link to View README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/view#readme)
- [ ] Workflows: [Link to Workflows README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/workflows#readme)

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [ ] Endpoint 1
- [ ] Endpoint 2

**_ARE THERE ANY RELATED PRs?_**

- [ ] Related PR 1
- [ ] Related PR 2

### CODE REVIEW

File: src/analytics/api/models/base/data_processing.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                                         pm_by_city,pm_by_country,
+                                         pm_by_region, monthly_mean_pm_site_name,
+                                         datetime_pm2_5, mean_pm2_5_by_month_year, 
+                                         mean_pm2_5_by_month_name, query_bigquery, 
+                                         mean_pm2_5_by_month, results_to_dataframe,
+                                           mean_pm2_5_by_year, mean_daily_pm2_5, 
+                                           mean_pm2_5_by_site_name, mean_pm2_5_by_hour)
+ 
+
+app = Flask(__name__)
```
Comment (Baalmart) [inline src/analytics/api/models/base/data_processing.py:14]: Thanks @wabinyai for this.
Just a quick clarification, do we need to start a new instance of the Flask App?

File: src/analytics/api/utils/pollutants/report.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    grid_url = f"https://platform.airqo.net/api/v2/devices/measurements/grids/{grid_id}"
+
+    try:
+        grid_response = requests.get(grid_url, params=grid_params)
+        grid_response.raise_for_status()  # Raise an exception for 4xx and 5xx status codes
+
+        data = grid_response.json()
+
+        # Extracting only the 'site_id' from each measurement
```
Comment (Baalmart) [inline src/analytics/api/utils/pollutants/report.py:32]: Hi @wabinyai , if we are only interested in the site_id from the Grids measurement API...wont it be more optimal to just then use the Grids summary endpoint (`{{baseUrl}}/api/v2/devices/grids/summary`) which just returns site IDs and other small details associated with the Grid.

 Ideally and generally, most endpoint for measurements take more time than others which just return an entity's details.

Additionally, this Grids Summary endpoint is NOT protected so you can just call it WITHOUT a token. :)

**Review by Baalmart — COMMENTED** [review]
Thanks @wabinyai , I have just left some code review comments for your review. I will test out the PR later this weekend.

File: src/analytics/api/models/base/data_processing.py (hunk 1 above)
Comment (wabinyai) [reply src/analytics/api/models/base/data_processing.py:14]: thanks for the keen eyes

**Review by BenjaminSsempala — APPROVED** [review]

File: src/analytics/api/utils/pollutants/report.py (hunk 2 above)
Comment (wabinyai) [reply src/analytics/api/utils/pollutants/report.py:32]: It will be resolved. On my end summary requests for token

**Review by Baalmart — APPROVED** [review]

**Pull request merged**